file-manager/fileManager.cpp

```diff
@@ -0,0 +1,168 @@
+            return false;
+        }
+        
+        // Write the binary data to the file
+        file.write(reinterpret_cast<const char*>(binaryData.bytes), binaryData.size);
+        file.close();
+
+        std::cout << "Download sucessful for: " << fileName << " at " << destiationFolder << std::endl;
+        return true;
+    } 
+    else 
+    {
+        std::cout << "File not found in the collection: " << fileName << std::endl;
+        return false;
+    }
+
+}
+
+
+// ********************************************** Main **********************************************
+int main()
+{
+    try
+    {
+        auto mongoURIStr = "<Insert MongoDB Connection String>";
+        static const mongocxx::uri mongoURI = mongocxx::uri{ mongoURIStr };
+
+        if(mongoURI.to_string().empty())
+        {
+            std::cout << "URI is empty";
+            return 0;
+        }
+        
+        // Create an instance.
+        mongocxx::instance inst{};
+        
+        mongocxx::options::client client_options;
+        auto api = mongocxx::options::server_api{ mongocxx::options::server_api::version::k_version_1 };
+        client_options.server_api_opts(api);
+        mongocxx::client conn{ mongoURI, client_options};
+        
+        const std::string dbName = "fileStorage";
+        const std::string collName = "files";
+        
+        auto fileStorageDB = conn.database(dbName);
+        auto filesCollection = fileStorageDB.collection(collName);
+
+        // Upload all files in the upload folder.
+        const std::string uploadFolder = "/Users/bishtr/repos/fileStorage/upload/";
+        for (const auto & filePath : std::filesystem::recursive_directory_iterator(uploadFolder))
+        {
+            if(std::filesystem::is_directory(filePath))
+                continue;
+
+            if(!upload(filePath.path().string(), filesCollection))
+            {
+                std::cout << "Upload failed for: " << filePath.path().string() << std::endl;
+            }
+        }
+
+        // Download files to the download folder.
+        const std::string downloadFolder = "/Users/bishtr/repos/fileStorage/download/";
+        
+        // Search with specific filenames and download it.
+        const std::string fileName1 = "image-15.jpg", fileName2 = "Hi Seed Shaker 120bpm On Accents.wav";
+        for ( auto fileName : {fileName1, fileName2} )
+        {
+            download(fileName, downloadFolder, filesCollection);
+        }
+        
+        // Download all files in the collection.
+        auto cursor = filesCollection.find({});
+        for (auto&& doc : cursor) 
+        {
+            auto fileName = doc[FILE_NAME].get_string().value.to_string();
+            download(fileName, downloadFolder, filesCollection);
+        }
+    }
+    
+    catch(const std::exception& e)
+    {
+        std::cout << "Exception encountered: " e.what() << std::endl;
```

Comment:
```suggestion
        std::cout << "Exception encountered: " << e.what() << std::endl;
```

Resolves compiler error.